Plugins/MatrixPlugin/MatrixAction.cs

```diff
@@ -0,0 +1,238 @@
+            {
+                _defenceAction.Attribute = value;
+                OnPropertyChanged();
+            }
+        }
+
+        [XmlIgnore]
+        public int DefenceModifier
+        {
+            get => _defenceAction.Modifier;
+            set
+            {
+                _defenceAction.Modifier = value;
+                OnPropertyChanged();
+            }
+        }
+
+        private Test tests;
+        [XmlElement("test")]
+        public Test Tests
+        {
+            get => tests;
+            set
+            {
+                tests = value;
+                Description = value.bonusstring;
+                ParseTest(value);
+            }
+        }
+
+        #endregion
+
+        private void ParseTest(Test test)
+        {
+            string[] SkillCheck = test.dice.Split('.');
+            //Parsing SkillCheck as
+            //Skill + Attribute|MatrixAttribute + Modifier vs. Skill + MatrixAttribute + Modifier
+            string actionCheck = SkillCheck[0];
+            string defenceCheck = SkillCheck.Length > 1 ? SkillCheck[1] : "";
+            //Parse Skills
+            foreach (string skill in SkillNames)
+            {
+                if (actionCheck.Contains(skill))
+                    ActionSkill = skill;
+                if (defenceCheck.Contains(skill))
+                    DefenceSkill = skill;
+            }
+            //Parse MatrixAttributes
+            foreach (string skill in MatrixAttributes.MatrixAttributeStrings)
+            {
+                if (!string.IsNullOrEmpty(defenceCheck) && defenceCheck.Contains(skill))
+                    DefenceSkill = skill;
+                if (!string.IsNullOrEmpty(test.limit) && test.limit.Contains(skill))
+                    Limit = skill;
+            }
+            //Parse MentalAttributes
+            foreach (string attr in AttributeSection.MentalAttributes)
+            {
+                if (!string.IsNullOrEmpty(actionCheck) && actionCheck.Contains(attr))
+                    ActionAttribute = attr;
+                if (!string.IsNullOrEmpty(defenceCheck) && defenceCheck.Contains(attr))
+                    DefenceAttribute = attr;
+            }
+            //Parse Modifiers
+            // as [-] Digits
+            if (!string.IsNullOrEmpty(actionCheck) && Regex.IsMatch(actionCheck, _dicePoolModifierRegexPattern))
+            {
+                Match match = Regex.Match(actionCheck, _dicePoolModifierRegexPattern);
+                string result = match.Groups[1].Value + match.Groups[2].Value;
+                ActionModifier = int.Parse(result);
+            }
+            if (!string.IsNullOrEmpty(defenceCheck) && Regex.IsMatch(defenceCheck, _dicePoolModifierRegexPattern))
+            {
+                Match match = Regex.Match(defenceCheck, _dicePoolModifierRegexPattern);
+                string result = match.Groups[1].Value + match.Groups[2].Value;
+                DefenceModifier = int.Parse(result);
+            }
```

Comment:
This is fine for very simple arithmetic like adding (or subtracting) a single modifier. However, if you want to expand this to support more complicated operations, it might be a better idea to have a look at `CommonFunctions.EvaluateInvariantXPath()` in the base Chummer project. C#'s XPath processor can actually handle quite a bit of arithmetic and basic logic gates. We use it all the time in the base Chummer project for when we need to evaluate some bit of arithmetic, but we can't know ahead of time what that arithmetic might be (other than that it might include multiplication and division, too).